plugins/node/instrumentation-amqplib/src/amqplib.ts

```diff
@@ -0,0 +1,749 @@
+        moduleExports.Channel.prototype,
+        'emit',
+        this.getChannelEmitPatch.bind(this)
+      );
+    }
+    if (!isWrapped(moduleExports.ConfirmChannel.prototype.publish)) {
+      this._wrap(
+        moduleExports.ConfirmChannel.prototype,
+        'publish',
+        this.getConfirmedPublishPatch.bind(this, moduleVersion)
+      );
+    }
+    return moduleExports;
+  }
+
+  private unpatchChannelModel(moduleExports: any) {
+    if (isWrapped(moduleExports.Channel.prototype.publish)) {
+      this._unwrap(moduleExports.Channel.prototype, 'publish');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.consume)) {
+      this._unwrap(moduleExports.Channel.prototype, 'consume');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.ack)) {
+      this._unwrap(moduleExports.Channel.prototype, 'ack');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.nack)) {
+      this._unwrap(moduleExports.Channel.prototype, 'nack');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.reject)) {
+      this._unwrap(moduleExports.Channel.prototype, 'reject');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.ackAll)) {
+      this._unwrap(moduleExports.Channel.prototype, 'ackAll');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.nackAll)) {
+      this._unwrap(moduleExports.Channel.prototype, 'nackAll');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.emit)) {
+      this._unwrap(moduleExports.Channel.prototype, 'emit');
+    }
+    if (isWrapped(moduleExports.ConfirmChannel.prototype.publish)) {
+      this._unwrap(moduleExports.ConfirmChannel.prototype, 'publish');
+    }
+    return moduleExports;
+  }
+
+  private getConnectPatch(
+    original: (
+      url: string | amqp.Options.Connect,
+      socketOptions: any,
+      openCallback: (err: any, connection: amqp.Connection) => void
+    ) => amqp.Connection
+  ) {
+    return function patchedConnect(
+      this: unknown,
+      url: string | amqp.Options.Connect,
+      socketOptions: any,
+      openCallback: Function
+    ) {
+      return original.call(
+        this,
+        url,
+        socketOptions,
+        function (this: unknown, err, conn: amqp.Connection) {
+          if (err === null) {
+            const urlAttributes = getConnectionAttributesFromUrl(url);
+            // the type of conn in @types/amqplib is amqp.Connection, but in practice the library send the
+            // `serverProperties` on the `conn` and not in a property `connection`.
+            // I don't have capacity to debug it currently but it should probably be fixed in @types or
+            // in the package itself
+            // currently setting as any to calm typescript
+            const serverAttributes = getConnectionAttributesFromServer(
+              conn as any
+            );
+            Object.defineProperty(conn, CONNECTION_ATTRIBUTES, {
+              value: { ...urlAttributes, ...serverAttributes },
+              enumerable: false,
```

Comment:
```suggestion
              enumerable: false, configurable: true, writable: true
```
at least `configurable: true` should be added as non configureable properties are quite uncommon.

Reply:
These attributes are actually const once calculated and should not be altered at any time in the future.
So it does make sense logically.
But if it causes performance or readability concerns I am good with changing it.

Reply:
As I'm coming from strong typed languages I agree. But JS is different as everything is mutable on default.
Just consider that everything what should not be altered would be really readonly/non configurable - not a single instrumentation would be possible.